qemu/tests/block_resize.py

```diff
@@ -9,6 +9,166 @@
 from virttest import data_dir
 
 
+def get_block_size(test, session, block_cmd, block_pattern):
+    """
+    Get the disk size inside guest.
+
+    param block_cmd: run block_cmd command to get detail disk size info.
+    param block_pattern: use this pattern to filter the needed.
+
+    param return: return current block size.
+    """
+    output = session.cmd_output(block_cmd)
+    block_size = re.findall(block_pattern, output)
+    if block_size:
+        if not re.search("[a-zA-Z]", block_size[0]):
+            return int(block_size[0])
+        else:
+            return float(utils_misc.normalize_data_size(block_size[0],
+                                                        order_magnitude="B"))
+    else:
+        test.error("Can not find the block size for the device."
+                   " The output of command is: %s" % output)
+
+
+def compare_block_size(test, session, params, expected_size, current_size):
+    """
+    Compare the current block size with the expected size.
+
+    param expected_size: the expected size after block resize.
+    param current_size: the size from guest after block resize.
+    """
+    accept_ratio = float(params.get("accept_ratio", 0))
+    if (current_size <= expected_size and
+            current_size >= expected_size * (1 - accept_ratio)):
+        logging.info("Block Resizing Finished !!! \n"
+                     "Current size %s is same as the expected %s",
+                     current_size, expected_size)
+        return True
+    else:
+        test.fail("Block size get from guest is not same as expected.\n"
+                  "Reported: %s\nExpect: %s\n" % (current_size, expected_size))
+    return
+
+
+def get_drive_path_linux(test, session, params, data_image):
+    """
+    Get the drive id in linux guest.
+
+    param data_image: data image name, like image1 or stg.
+    param return: return drive path, like /dev/sda or /dev/sdb.
+    """
+    pattern = r"(serial|wwn)=(\w+)"
+    match = re.search(pattern, params["blk_extra_params_%s" % data_image], re.M)
+    if match:
+        drive_id = match.group(2)
+    else:
+        test.fail("No available tag to get drive id")
+    drive_path = utils_misc.get_linux_drive_path(session, drive_id)
+    if not drive_path:
+        test.error("Failed to get '%s' drive path" % data_image)
+    return drive_path
+
+
+def block_resize_enlarge(vm, session, params, device, block_size):
+    """
+    Enlarge the disk image of device to expected block_size.
+
+    param device: enlarge device.
+    param block_size: enlarge device to block_size.
+    """
+    logging.info("Enlarge disk size to %s in monitor"
+                 % block_size, logging.info)
+    vm.monitor.block_resize(device, block_size)
+    pattern = params.get("block_size_pattern")
+    block_size_cmd = params["block_size_cmd"]
```

Comment:
useless variables "block_size_cmd" and "pattern" in this function.

Reply:
will delete them, thanks~